xdsl/dialects/affine.py

```diff
@@ -52,10 +53,10 @@ def verify_(self) -> None:
 
     @staticmethod
     def from_region(operands: list[Operation | SSAValue],
-                    lower_bound: int | IntegerAttr,
-                    upper_bound: int | IntegerAttr,
+                    lower_bound: int | IntegerAttr[Any],
+                    upper_bound: int | IntegerAttr[Any],
```

Comment:
AnyIntegerAttr should be a better shorthand (I think it also constraints the dependent parameter)